### PR TITLE
Reject Sigma Double Mint

### PR DESCRIPTION
**Pull request by levonpetrosyan93**

### CODE REVIEW

File: src/zerocoin_v3.cpp [hunk 1]
```diff
@@ -355,6 +355,10 @@ bool CheckSigmaMintTransaction(
     if (hasCoin) {
        LogPrintf("CheckSigmaMintTransaction: double mint, tx=%s\n",
                 txout.GetHash().ToString());
+        return state.DoS(100,
```
Comment (martun) [inline src/zerocoin_v3.cpp]: Are we sure we want function CheckSigmaMintTransaction to fail, in case it's called twice?

Comment (levonpetrosyan93) [reply src/zerocoin_v3.cpp]: Yes I see this function can be called twice, it can be at miner. let me add a condition before return.

**Comment by ultimaweapon** [comment]
Sometime ago I found it called twice but I cannot find a second reference anymore. Not sure which way we should go.

**Comment by psolstice** [comment]
I think there is a need for `fStatefulCheck` flag here. So the double mint is checked only if it is set to true

**Pull request merged**